src/main/java/org/wso2/carbon/connector/connection/EmailConnectionPool.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector.connection;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.pool.impl.GenericObjectPool;
+import org.wso2.carbon.connector.exception.EmailConnectionPoolException;
+import org.wso2.carbon.connector.pojo.ConnectionConfiguration;
+
+import static java.lang.String.format;
+
+/**
+ * Connection Pool Implementation for Email Connections
+ */
+public class EmailConnectionPool extends GenericObjectPool {
```

Comment:
Is it possible to move connection pool related classes to a different package? Then in the long run we can evaluate the possibility to move the implementation to the connector core with required interfaces.

Reply:
Moved to connector core